pkg/kubelet/kubelet_node_status.go

```diff
@@ -298,7 +298,12 @@ func (kl *Kubelet) setNodeAddress(node *api.Node) error {
 		if err != nil {
 			return fmt.Errorf("failed to get node address from cloud provider: %v", err)
 		}
-		node.Status.Addresses = nodeAddresses
+		for _, address := range nodeAddresses {
+			if address.Type == api.NodeHostName {
+				return fmt.Errorf("HostnameAddress should not be set by cloudprovider.")
```

Comment:
it seems like the cloud provider would be best positioned to know the hostname in some cases (e.g. the AWS provider querying "public-hostname" from metadata). I agree the interaction with user-specified --hostname-override is weird, I just don't know that the cloudprovider should be forbidden from resolving a hostname